### PR TITLE
Allow prerendering NotFoundPage as 404.html

### PR DESCRIPTION
**Pull request by dac09**
## What?
- Allows generation of 404 pages from NotFoundPage in Routes.js/tsx
- Also changes defaultIndex.html -> 200.html to follow same convention

--- 
**Why 404.html?**
https://docs.netlify.com/routing/redirects/redirect-options/#custom-404-page-handling

### CODE REVIEW

**Comment by thedavidprice** [comment]
Curious if we should add this as a default setup for new projects (i.e. prerender NotFoundPage) —  is there a downside?

**Comment by dac09** [comment]
> Curious if we should add this as a default setup for new projects (i.e. prerender NotFoundPage) — is there a downside?

Only downside is they're not prepared for their NotFoundPage to be prerendered. Maybe they use a library that doesn't work in Node.js?

File: packages/prerender/src/__tests__/detectRoutes.test.ts [hunk 1]
```diff
@@ -50,18 +50,20 @@ describe('Detecting routes', () => {
     )
   })
 
-  it('Should ignore not found page', () => {
+  it('Should render notFoundPage as 404.html', () => {
     mockedRoutes = [
       {
-        name: 'bazinga',
-        path: '/bazinga',
+        name: undefined,
```
Comment (dac09) [inline packages/prerender/src/__tests__/detectRoutes.test.ts:56]: This is how structure outputs NotFoundPages

**Comment by thedavidprice** [comment]
> Only downside is they're not prepared for their NotFoundPage to be prerendered. Maybe they use a library that doesn't work in Node.js?

Understood that could be unexpected and hard to diagnose, especially given that at this time there would be no errors locally but only at the time of deploy. Even though it wouldn't be breaking, definitely not ideal.

This gives me a couple of ideas I'll table for now. Thanks and carry on!

File: packages/prerender/src/__tests__/detectRoutes.test.ts (hunk 1 above)
Comment (jvanbaarsen) [reply packages/prerender/src/__tests__/detectRoutes.test.ts:56]: would `path: "some-unknown-path"` not be more clearly indicate intent? Or doesn't that work in this case?

Comment (dac09) [reply packages/prerender/src/__tests__/detectRoutes.test.ts:56]: We're mocking the output of the structure package here. This is just what it outputs, when it comes across a route with a not found prop.

```
      <Route notfound page={NotFoundPage} prerender />
```

The code that deals with that actually doesn't really care about the values of name/path, i've just made this change to more accurately represent what structure does.

File: packages/prerender/src/runPrerender.tsx [hunk 2]
```diff
@@ -81,10 +81,10 @@ export const runPrerender = async ({
   }
 
   if (outputHtmlPath) {
-    // Copy default index.html to defaultIndex.html first
+    // Copy default index.html to 200.html first
```
Comment (peterp) [inline packages/prerender/src/runPrerender.tsx:84]: 🥇

**Review by peterp — APPROVED** [review]

**Pull request merged**